cmd/vclusterctl/cmd/create.go

```diff
@@ -269,16 +277,44 @@ func (cmd *CreateCmd) deployChart(vClusterName, chartValues, helmExecutablePath
 		return fmt.Errorf("aborting vcluster creation. Current working directory contains a file or a directory with the name equal to the vcluster chart name - \"%s\". Please execute vcluster create command from a directory that doesn't contain a file or directory named \"%s\"", cmd.ChartName, cmd.ChartName)
 	}
 
-	// rewrite chart location, this is an optimization to avoid
-	// downloading the whole index.yaml and parsing it
-	if cmd.LocalChartDir == "" && cmd.ChartVersion != "" && cmd.ChartRepo == LoftChartRepo {
-		if cmd.ChartVersion[0] == 'v' {
-			cmd.ChartVersion = cmd.ChartVersion[1:]
-		}
+	if cmd.LocalChartDir == "" {
+		chartEmbedded := false
+		if cmd.ChartVersion == upgrade.GetVersion() { // use embedded chart if default version
+			embeddedChartName := fmt.Sprintf("%s-%s.tgz", cmd.ChartName, upgrade.GetVersion())
+			// not using filepath.Join because the embed.FS separator is not OS specific
+			embeddedChartPath := fmt.Sprintf("charts/%s", embeddedChartName)
+
+			embeddedChartFile, err := Charts.ReadFile(embeddedChartPath)
+			if err != nil && errors.Is(err, fs.ErrNotExist) {
+				cmd.log.Infof("Chart not embedded: %q, pulling from helm repository.", err)
+			} else if err != nil {
+				cmd.log.Errorf("Unexpected error while accessing embedded file: %q", err)
+			} else {
+				temp, err := os.CreateTemp("", fmt.Sprintf("%s%s", embeddedChartName, "-"))
+				if err != nil {
+					cmd.log.Errorf("Error creating temp file: %v", err)
+				} else {
+					defer temp.Close()
+					defer os.Remove(temp.Name())
+				}
 
-		cmd.LocalChartDir = LoftChartRepo + "/charts/" + cmd.ChartName + "-" + cmd.ChartVersion + ".tgz"
-		cmd.ChartVersion = ""
-		cmd.ChartRepo = ""
+				_, err = temp.Write(embeddedChartFile)
```

Comment:
Please move this as well as the code below into the else statement as this can be nil is os.CreateTemp fails

Reply:
fixued